subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -447,10 +447,27 @@ static int cmd_wifi_disconnect(const struct shell *sh, size_t argc,
 static int cmd_wifi_scan(const struct shell *sh, size_t argc, char *argv[])
 {
 	struct net_if *iface = net_if_get_default();
+	struct wifi_scan_params params = { 0 };
 
 	context.sh = sh;
 
-	if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, NULL, 0)) {
+	if (argc > 2) {
+		shell_fprintf(sh, SHELL_WARNING, "Invalid number of arguments\n");
+		return -ENOEXEC;
+	}
+
+	if (argc == 2) {
```

Comment:
As the command now takes (optional) parameters, the help string in https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/wifi/wifi_shell.c#L1109 should be updated.